docs/api/performance.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: Performance Optimization
+---
+
+Achieving the best performance out of ParadeDB requires configuring it properly for your system
+and workload(s). Here we provide a list of best practices to serve as a starting point to optimizing
+your search queries.
+
+## Implement Pagination
+
+If you do not specify any `limit_rows` and `offset_rows` ParadeDB will rank results on your entire
+index, which can meaningfully slow down search queries. We suggest that you use limits and offsets to
+scope each search query to the number of results that you want to display at any given time on your application.
+
+Note the use of `limits_rows` and `offset_rows`, as opposed to `LIMIT` and `OFFSET`, which are not pushed down
+to the BM25 index.
+
+```sql
+-- ✅ Correct syntax
+SELECT description, rating, category
+FROM search_idx.search(
+  '(description:keyboard OR category:electronics) AND rating:>2',
+  limit_rows => 5
+);
+
+-- ❌ Incorrect syntax
+SELECT description, rating, category
+FROM search_idx.search(
+  '(description:keyboard OR category:electronics) AND rating:>2')
+LIMIT 5;
+```
+
+## Select the Appropriate Hardware
+
+Higher amounts of RAM and a faster storage driver are the two most important hardware components
+to extracting the best performance from ParadeDB. We recommend that you select memory-optimized instances
+with SSD storage when deploying.
+
+In terms of memory, ParadeDB requires at least 500 MB of RAM at any given time for the writer process. Higher
+amounts of memory will allow storing a larger result set without needing to read from disk, which will speed
+up search queries.
+
+In terms of storage, faster I/O storage devices like `gp3` will provide better performance. You can expect a
+search index over a table to be at least 50% of the size of the indexed table. The exact index size will vary
+based on indexing parameters like tokenization scheme.
+
+## Optimize the Tantivy Settings
+
+Coming soon.
+
+### Heap Size
+
+The memory_budget setting defines the maximum heap size in bytes allocated to a search index. Proper configuration
+ensures efficient memory use, preventing excessive disk I/O during searches.
+
+Default: 500 MB (500_000_000 bytes)
+Minimum: 15 MB (15_000_000 bytes)
+
+Adjust the memory_budget based on your system's available memory and the size of your indexes to optimize performance.
+For most systems, 10-20MB per indexer thread is maximum. We get limited improvement after this.
+
+### Reader Cache
+
+The reader_cache_num_blocks setting controls the number of blocks reserved for caching index data in memory. A well-sized
+reader cache minimizes disk access, leading to faster query responses.
+
+Default: 100 blocks
+
+Typically works best between 512-1024 documents.
+
```

Comment:
@vaibhawvipul Do you recall what you meant by this? Why is default in blocks unit vs suggestion in documents unit?